x/fswap/types/genesis.go

```diff
@@ -53,5 +41,24 @@ func (gs *GenesisState) Validate() error {
 		return ErrInvalidState.Wrap("number of swaps does not match swap count in SwapStats")
 	}
 
+	swappeds := gs.GetSwappeds()
+	for i, swap := range gs.GetSwaps() {
+		swapped := swappeds[i]
+		if err := swap.ValidateBasic(); err != nil {
+			return err
+		}
+		if err := swapped.ValidateBasic(); err != nil {
+			return err
+		}
+		if swap.FromDenom != swapped.FromCoinAmount.Denom {
+			return ErrInvalidState.Wrap("FromCoin in swap and swapped do not correspond")
+		}
+		if swap.ToDenom != swapped.ToCoinAmount.Denom {
+			return ErrInvalidState.Wrap("ToCoin in swap and swapped do not correspond")
+		}
+		if swap.AmountCapForToDenom.LT(swapped.ToCoinAmount.Amount) {
+			return ErrInvalidState.Wrap("AmountCapForToDenom cannot be exceeded")
```

Comment:
This brings another restriction. 
It prohibits a chain from swap-update to increase/decrease `AmountCapForToDenom` regardless of the state of accumulated `swapped.ToCoinAmount.Amount`.

When we may want to stop swap, we could update AmountCapForToDenom to zero. but with this validation chain run fails.

I'm not sure we need this restriction. 
If we need, there is check logic not to accept amount `AmountCapForToDenom` which is smaller then current `swapped.ToCoinAmount.Amount` in update logic.

Reply:
@0Tech Could you please check this part?

> When we may want to stop swap, we could update AmountCapForToDenom to zero.
I think this is not a good way to do that. In my opinion, `Swaps` could not been modified once it is set. If there is a need to manually stop the swap, then I think adding an extra boolean field would be better.

Reply:
>When we may want to stop swap, we could update AmountCapForToDenom to zero.
It's only example usecase regarding to restriction of upgradable values.

I think, we don't know, what value should we set as upgradable or not upgradable.
Before adding this kind of restriction logic, we need enough use cases with some reasons.
I don't think, we have specific requirement or usecase to restrict for now.

Reply:
It's genesis logic, so we won't need to worry about possible race conditions. Then, we can actually set the cap to the current amount in case of stopping the swap. Hence, with this verification, we can still modify the swap to cover certain scenarios in the future. It seems reasonable to restrict the swapped to be lesser than or equal to the cap.

Reply:
@jaeseung-bae I agree with 0Tech
> When we may want to stop swap, we could update AmountCapForToDenom to zero. but with this validation chain run fails.

As 0Tech said, this could be done by setting the cap to the current amount.

Reply:
I want to check the spec again.

We can update `AmountCapForToDenom` value if config allows upgrade.
And we can only increase `AmountCapForToDenom` value because we don't allow cap is less then swapped amount `if swap.AmountCapForToDenom.LT(swapped.ToCoinAmount.Amount)`.

It this right?